modules/gapi/test/serialization/gsrlzbackend.cpp

```diff
@@ -0,0 +1,303 @@
+            //Graph serialization - dump - read - de-serialization path
+            //Dump graph file.
+            auto serialized_graph = cv::gimpl::serialization::serialize(graph, nodes);
+            std::ofstream dump_file;
+            cv::gimpl::serialization::SerializationStream out_stream;
+            cv::gimpl::serialization::dumpGSerialized(serialized_graph, out_stream);
+            CV_LOG_INFO(NULL, "out_stream.dump_storage size in bytes ..." << out_stream.getSize());
+            dump_file.open("my_graph.bin", std::ofstream::out | std::ofstream::trunc | std::ofstream::binary);
+            if (dump_file.is_open())
+            {
+                dump_file.seekp(0, std::ofstream::beg);
+                dump_file.write((const char*)out_stream.getData(), out_stream.getSize());
+            }
+            dump_file.close();
+            ////////////////////////////////////////////////////////////
+            //Restore graph from file.
+            std::ifstream new_dump_file ("my_graph.bin", std::ifstream::in | std::ofstream::binary);
+            cv::gimpl::serialization::GSerialized new_serialized_graph;
+            if (new_dump_file.is_open())
+            {
+                new_dump_file.seekg(0, new_dump_file.end);
+                std::streampos length = new_dump_file.tellg();
+                CV_LOG_INFO(NULL, "new_dump_file size in bytes ..." << (int)length);
+                new_dump_file.seekg(0, new_dump_file.beg);
+                char * buffer = new char[(int)length];
+                new_dump_file.read(buffer, length);
+                cv::gimpl::serialization::DeSerializationStream in_stream =
+                    cv::gimpl::serialization::DeSerializationStream(buffer, length);
+                delete[] buffer;
+                cv::gimpl::serialization::readGSerialized(new_serialized_graph, in_stream);
+            }
+            new_dump_file.close();
+
+            auto gp = std::make_shared<ade::Graph>();
+            auto& g_s = *gp.get();
+            std::vector<ade::NodeHandle> nh;
+
+            for (const auto& data : new_serialized_graph.m_datas)
+            {
+                cv::gimpl::serialization::mkDataNode(g_s, data);
+            }
+
+            for (const auto& op : new_serialized_graph.m_ops)
+            {
+                cv::gimpl::serialization::mkOpNode(g_s, op);
+            }
+
+            nh = cv::gimpl::serialization::linkNodes(g_s);
+            CV_LOG_INFO(NULL, "nh Size " << nh.size());
+
+            //Use CPU serialization kernels package to test
+            cv::gapi::GKernelPackage srlz_kernels = combine(cv::gapi::core::serialization::kernels(), cv::gapi::imgproc::serialization::kernels());
+            //Compiler pass one more time
+            auto pass_ctx = ade::passes::PassContext{g_s};
+            cv::gimpl::passes::resolveKernels(pass_ctx, srlz_kernels);
+
+            return EPtr{new cv::gimpl::GSRLZExecutable(g_s, nh, gp)};
```

Comment:
Oh I see why do you need this now. Well... Need to think about it. Please add a FIXME in the places you've exported

Reply:
Done. FIXMEs were added.